sdcm/utils/file.py

```diff
@@ -0,0 +1,109 @@
+                 closefd: bool = True):
+        self.path = path
+        self.mode = mode
+        self.buffering = buffering
+        self.encoding = encoding
+        self.errors = errors
+        self.newline = newline
+        self.closefd = closefd
+        self._io = self._open()
+
+    def get_file_length(self) -> int:
+        return self._open().seek(0, 2)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def _open(self) -> TextIO:
+        kwargs = {attr_name: getattr(self, attr_name) for attr_name in
+                  ['mode', 'buffering', 'encoding', 'errors', 'closefd'] if getattr(self, attr_name, None) is not None}
+        return open(self.path, **kwargs)  # deepcode ignore missing~close~open: <comment the reason here>
+
+    def move_to(self, pos) -> 'File':
+        self._io.seek(pos)
+        return self
+
+    def move_to_end(self) -> 'File':
+        self._io.seek(0, 2)
+        return self
+
+    def move_to_beginning(self) -> 'File':
+        self._io.seek(0)
+        return self
+
+    def move_to_relative(self, pos: int) -> 'File':
+        self._io.seek(pos, 1)
+        return self
+
+    def move_to_relative_from_end(self, pos: int = 0) -> 'File':
+        self._io.seek(pos, 2)
+        return self
+
+    def write(self, s: AnyStr) -> 'File':
+        self._io.write(s)
+        return self
+
+    def writelines(self, s: List[AnyStr]) -> 'File':
+        self._io.writelines(s)
+        return self
+
+    def seek(self, offset: int, whence: int = 0) -> 'File':
+        self._io.seek(offset, whence)
+        return self
+
+    def read(self, n: int = -1) -> AnyStr:
+        return self._io.read(n)
+
+    def readline(self, limit: int = -1) -> AnyStr:
+        return self._io.readline(limit)
+
+    def readlines(self, hint: int = -1) -> List[AnyStr]:
+        return self._io.readlines(hint)
+
+    def read_lines_filtered(self, patterns: Union[str, Pattern, DatabaseLogEvent], ignore_case=True) \
+            -> List[str]:
+        """
+        Read lines from the file, filter them and yield
+        :param patterns: List of patterns
+        :param ignore_case: If true, it will ignore cases when searching for a pattern
+        :return: List of strings that matched patterns
+        """
+        regexps = []
+        if ignore_case:
+            re_params = (IGNORECASE,)
+        else:
+            re_params = tuple()
+        for pattern in patterns:
+            if isinstance(pattern, Pattern):
+                regexps.append(pattern)
+            elif isinstance(pattern, str):
+                regexps.append(compile(pattern, *re_params))
+            elif isinstance(pattern, DatabaseLogEvent):
+                regexps.append(compile(pattern.regex, *re_params))
+        for line in self._io:
+            for pattern in regexps:
+                if pattern.search(line):
+                    yield line
+                    break
```

Comment:
Do you find that it'll be much better to extract this method to a separate function and drop the whole `File` class as a trivial wrapper around `open()`?